### PR TITLE
Mining laser fixes

### PR DESCRIPTION
**Pull request by HybridDog**

### CODE REVIEW

File: technic/tools/mining_lasers.lua [hunk 1]
```diff
@@ -49,24 +50,28 @@ local function laser_node(pos, node, player)
...
+
 local function laser_shoot(player, range, particle_texture, sound)
 	local player_pos = player:getpos()
 	local player_name = player:get_player_name()
 	local dir = player:get_look_dir()
 
 	local start_pos = vector.new(player_pos)
 	-- Adjust to head height
-	start_pos.y = start_pos.y + 1.6
+	start_pos.y = start_pos.y + 1.625
```
Comment (SmallJoker) [inline technic/tools/mining_lasers.lua]: Could make use of the 5.0.0 player attribute `eye_height` and use `1.625` as fallback value.
https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L5226

File: technic/tools/mining_lasers.lua [hunk 2]
```diff
@@ -49,24 +50,28 @@ local function laser_node(pos, node, player)
...
 	minetest.add_particle({
-		pos = startpos,
+		pos = start_pos,
 		velocity = dir,
 		acceleration = vector.multiply(dir, 50),
-		expirationtime = range / 11,
+		expirationtime = (math.sqrt(1+100*(range+0.4))-1)/50,
```
Comment (SmallJoker) [inline technic/tools/mining_lasers.lua]: Please add spaces around binary operators

File: technic/tools/mining_lasers.lua [hunk 3]
```diff
@@ -4,43 +4,44 @@ local mining_lasers_list = {
...
+		{"",                "technic:carbon_steel_ingot", "technic:blue_energy_crystal"},
+		{"",                "",                           "default:copper_ingot"},
 	}
 })
 
 local function laser_node(pos, node, player)
 	local def = minetest.registered_nodes[node.name]
-	if def and def.liquidtype ~= "none" then
+	if def.liquidtype ~= "none"
+	and def.buildable_to then
```
Comment (SmallJoker) [inline technic/tools/mining_lasers.lua]: Two indents for continued if-statement lines, please. Same for the other changes below.

File: technic/tools/mining_lasers.lua [hunk 4]
```diff
@@ -49,24 +50,28 @@ local function laser_node(pos, node, player)
 	minetest.node_dig(pos, node, player)
 end
 
-local no_destroy = {
-	["air"] = true,
-	["default:lava_source"] = true,
-	["default:lava_flowing"] = true,
-}
+local no_destroy = {air = true}
```
Comment (SmallJoker) [inline technic/tools/mining_lasers.lua]: Inversion is annoying to handle. Could you please either convert this to `keep_node` and the function to `can_keep_node` (or similar)?
Also: `ignore = true`. Below you use `get_node_or_nil` which can return `ignore` when the map is limited.

Comment (HybridDog) [reply technic/tools/mining_lasers.lua]: That's actually a wrongly used get_node_or_nil.

**Comment by HybridDog** [comment]
I did the requested changes.

File: technic/tools/mining_lasers.lua [hunk 5]
```diff
@@ -49,24 +49,28 @@ local function laser_node(pos, node, player)
...
+
 local function laser_shoot(player, range, particle_texture, sound)
 	local player_pos = player:getpos()
 	local player_name = player:get_player_name()
 	local dir = player:get_look_dir()
 
 	local start_pos = vector.new(player_pos)
 	-- Adjust to head height
-	start_pos.y = start_pos.y + 1.6
+	start_pos.y = start_pos.y + player:get_properties().eye_height
```
Comment (SmallJoker) [inline technic/tools/mining_lasers.lua]: `(player:get_properties().eye_height or 1.6)` for backwards compat

Comment (HybridDog) [reply technic/tools/mining_lasers.lua]: it's actually 1.625

**Pull request merged**